src/lib/components/Cytoscape.react.js

```diff
@@ -511,94 +507,166 @@ Cytoscape.propTypes = {
     // Common props
 
     /**
-     * A list of dictionaries representing the elements of the networks.
-     *     1. Each dictionary describes an element, and specifies its purpose.
-     *         - `group` (string): Either 'nodes' or 'edges'. If not given, it's automatically inferred.
-     *         - `data` (dictionary): Element specific data.
-     *              - `id` (string): Reference to the element, useful for selectors and edges. Randomly assigned if not given.
-     *              - `label` (string): Optional name for the element, useful when `data(label)` is given to a style's `content` or `label`. It is only a convention.
-     *              - `parent` (string): Only for nodes. Optional reference to another node. Needed to create compound nodes.
-     *              - `source` (string): Only for edges. The id of the source node, which is where the edge starts.
-     *              - `target` (string): Only for edges. The id of the target node, where the edge ends.
-     *         - `position` (dictionary): Only for nodes. The position of the node.
-     *              - `x` (number): The x-coordinate of the node.
-     *              - `y` (number): The y-coordinate of the node.
-     *         - `selected` (boolean): If the element is selected upon initialisation.
-     *         - `selectable` (boolean): If the element can be selected.
-     *         - `locked` (boolean): Only for nodes. If the position is immutable.
-     *         - `grabbable` (boolean): Only for nodes. If the node can be grabbed and moved by the user.
-     *         - `classes` (string): Space separated string of class names of the element. Those classes can be selected by a style selector.
-     *
-     *     2. The [official Cytoscape.js documentation](http://js.cytoscape.org/#notation/elements-json) offers an extensive overview and examples of element declaration.
-     * Alternatively, a dictionary with the format { 'nodes': [], 'edges': [] } is allowed at initialization, but arrays remain the recommended format.
+     * A list of dictionaries representing the elements of the networks. Each dictionary describes an element, and
+     * specifies its purpose. The [official Cytoscape.js documentation](https://js.cytoscape.org/#notation/elements-json)
+     * offers an extensive overview and examples of element declaration.
+     * Alternatively, a dictionary with the format { 'nodes': [], 'edges': [] } is allowed at initialization,
+     * but arrays remain the recommended format.
      */
-    elements: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
+    elements: PropTypes.oneOfType([
+        PropTypes.arrayOf(
+            PropTypes.exact({
```

Comment:
Do we want to use `exact` or would `shape` be nicer here? I guess it doesn't hurt to warn the user about unwanted props but in this case I don't think the original JS library intended to have a strict typing here. cc @alexcjohnson

Reply:
I agree, shape would be nicer if it's acceptable to include other props.   I did that in the `layout` prop where it was clear from your description that that was the case

Reply:
Ah, good catch! In cytoscape you can add random extra data to elements and you can reference that data for example in stylesheets. So `shape` actually would be preferable here.

Reply:
```suggestion
            PropTypes.shape({
```